### PR TITLE
Vks4git/309/constraint error

### PR DESCRIPTION
**Pull request by vks4git**

### CODE REVIEW

File: symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs [hunk 1]
```diff
@@ -53,19 +54,19 @@ instance (Ord a, Arbitrary a, Binary a, KnownNat i) => Arbitrary (PlonkConstrain
 
 toPlonkConstraint :: forall a i . (Ord a, FiniteField a, KnownNat i) => Poly a (Var a (Vector i)) Natural -> PlonkConstraint i a
 toPlonkConstraint p =
-    let xs    = toList (variables p)
+    let xs    = Just <$> toList (variables p)
         perms = nubOrd $ map (take 3) $ permutations $ case length xs of
-            0 -> [ConstVar one, ConstVar one, ConstVar one]
```
Comment (TurtlePU) [inline symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs:58]: Is this change needed?

File: symbolic-base/test/Tests/Plonkup.hs [hunk 2]
```diff
@@ -33,10 +36,23 @@ import           ZkFold.Base.Protocol.Plonkup.Utils                  (sortByList
...
 
--- TODO: uncomment after refactoring
--- propPlonkConstraintConversion :: (Eq a, FiniteField a) => PlonkConstraint 1 a -> Bool
--- propPlonkConstraintConversion p =
---     toPlonkConstraint (fromPlonkConstraint p) == p
+-- | Polynomial types and specific polynomials that were causing exceptions
+--
+problematicPolynomials :: (Ord a, FiniteField a) => [PM.Poly a (Var a (Vector 1)) Natural]
+problematicPolynomials =
+    [ var (ConstVar one)
```
Comment (TurtlePU) [inline symbolic-base/test/Tests/Plonkup.hs:43]: These look like a part of the `ConstVar` test

File: symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs (hunk 1 above)
Comment (vks4git) [reply symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs:58]: I do not see how else we can separate `ConstVar one` here as a placeholder for an impossible variable and `ConstVar one` as an input polynomial. `perms` will hold exactly the same value if the input polynomial does not have any variables and if it is `ConsyVar one`

Comment (TurtlePU) [reply symbolic-base/src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs:58]: But do we need to separate a placeholder from an input polynomial?

**Review by TurtlePU — APPROVED** [review]
